cachi2/core/package_managers/npm.py

```diff
@@ -0,0 +1,226 @@
+import json
+import logging
+import os
+from pathlib import Path
+from typing import Any, Dict, Iterator, List
```

Comment:
Nitpick: you don't need `Dict` and `List`, the lowercase variants take type params as well ([since py3.9](https://peps.python.org/pep-0585/))